collections.go

```diff
@@ -1,4 +1,122 @@
 package outline
 
+import (
+	"context"
+	"fmt"
+
+	"github.com/dghubble/sling"
+	"github.com/ioki-mobility/go-outline/internal/common"
+)
+
 // CollectionsClient exposes CRUD operations around the collections resource.
-type CollectionsClient struct{}
+type CollectionsClient struct {
+	sl *sling.Sling
+}
+
+func newCollectionsClient(sl *sling.Sling) *CollectionsClient {
+	return &CollectionsClient{sl: sl}
+}
+
+func (cl *CollectionsClient) Get(id CollectionID) *CollectionsGetClient {
+	return newCollectionsGetClient(cl.sl, id)
+}
+
+func (cl *CollectionsClient) List() *CollectionsListClient {
+	return newCollectionListClient(cl.sl)
+}
+
+type CollectionsGetClient struct {
+	sl *sling.Sling
+}
+
+func newCollectionsGetClient(sl *sling.Sling, id CollectionID) *CollectionsGetClient {
+	data := struct {
+		ID CollectionID `json:"id"`
+	}{ID: id}
+
+	copy := sl.New()
```

Comment:
Just to get it right, this will "copy" the existing setup of `Sling` (therefore, copy headers and stuff), correct?